lib/zipper.ex

```diff
@@ -58,12 +58,12 @@ defmodule Styler.Zipper do
   @doc """
   Returns a new branch node, given an existing node and new children.
   """
-  @spec make_node(tree, [tree]) :: tree
-  def make_node({form, meta, _}, args) when is_atom(form), do: {form, meta, args}
-  def make_node({_form, meta, args}, [first | rest]) when is_list(args), do: {first, meta, rest}
-  def make_node({_, _}, [left, right]), do: {left, right}
-  def make_node({_, _}, args), do: {:{}, [], args}
-  def make_node(list, children) when is_list(list), do: children
+  @spec replace_children(tree, [tree]) :: tree
```

Comment:
I feel like it's worth noting that this also takes a `tree` and returns a `tree` instead of working with a `zipper` in the first argument, which means you ended up having to use it like this:

```elixir
# parent is a zipper, directives is a list of AST nodes
Zipper.update(parent, &Zipper.replace_children(&1, directives))
```

instead of just like this:

```elixir
# parent is a zipper, directives is a list of AST nodes
Zipper.replace_children(parent, directives)
```

I'm not really sure if this is the right thing to do or not, but what if we had the `Zipper` public API only operate on zippers (except for `Zipper.zip` to bootstrap it), and the existing `make_node` would be come a private function that's only used internally in a couple of places for shuffling things around? 🤷

Reply:
Actually, can't you implement `replace_children` on your own without a special API, like:

```elixir
# parent is a zipper, directives is a list of AST nodes
Zipper.update(parent, fn {node_name, meta, _children} -> {node_name, meta directives} end)
```

Reply:
>Actually, can't you implement replace_children on your own without a special API, like:

kind of? it gets weird because `{a, b, c}` isn't the only possible shape of a parent `Macro.t()` -- have a look at `replace_children`'s implementation to see all the weirdness

there is a clause in there i don't get, that does `{:{}, ...}` when there's a twople, and there are situations where i have to manually replace the parent with `{:__block__` instead (when adding more expressions to what was a single child expression situation  -- i do that both in this module and in the pipes module). sooo yeah, future stuff to keep an eye out for

Reply:
Yeah I meant specifically where you're doing it in `ModuleDirectives`, I think you _would_ know that it's a 3-tuple, right? I was saying maybe Zipper doesn't actually need a generic function to do this because you can just do it directly with `Zipper.update` and we could defer adding a new helper function until we see that it makes sense.

Reply:
> Yeah I meant specifically where you're doing it in ModuleDirectives, I think you would know that it's a 3-tuple, right? I

you do when it's a `defmodule`, but i'm less sure about other places that you can randomly add an `alias` to elixir code.

probably it's safe to assume that though?

Reply:
for now i'll leave it as is though, just to get this in